attributes/config.rb

```diff
@@ -79,11 +79,14 @@
   end
 
 if node.key?('hadoop') && node['hadoop'].key?('distribution') && node['hadoop'].key?('distribution_version') &&
-   node['hadoop']['distribution'] == 'hdp' && node['hadoop']['distribution_version'].to_f >= 2.2 &&
-   node['cdap']['version'].to_f >= 3.1
-  default['cdap']['cdap_env']['opts'] = "${OPTS} -Dhdp.version=#{hdp_version}"
-  default['cdap']['cdap_env']['spark_home'] = "/usr/hdp/#{hdp_version}/spark"
-  default['cdap']['cdap_site']['app.program.jvm.opts'] = "-XX:MaxPermSize=128M ${twill.jvm.gc.opts} -Dhdp.version=#{hdp_version} -Dspark.yarn.am.extraJavaOptions=-Dhdp.version=#{hdp_version}"
-else
+   node['hadoop']['distribution'] == 'hdp' && node['hadoop']['distribution_version'].to_f >= 2.2
+  if node['cdap']['version'].to_f >= 3.1
```

Comment:
unless rubocop says otherwise, there's no reason to split this into its own conditional rather than the previous compound stmt.  in fact, we can probably remove it altogether (in favor of an explicit compatability check).